### ISSUE TITLE
Investigate using Virtus/ROM to replace some core code

### ISSUE THREAD

**Issue by tokengeek** [fog/fog-core#5]
`fog-core` has a DSL for describing attributes on models and mapping the data responses between API responses and those attributes.

I assume @geemus added this because nothing was around that did this or it seemed a simple problem at the time.

So with the goal of cutting down on code we have to rewrite tests for (as we move away from Shindo) and balancing the new dependencies between less code to maintain I had been thinking about using Virtus to replace the code in https://github.com/fog/fog-core/blob/master/lib/fog/core/attributes.rb (but keep confusing it with ROM).

However the fog DSL also covers mapping so:

```
attribute :state, :aliases => 'instanceState', :squash => 'name'
```

So `state` is set from...

```
{
  "instanceState" => {
    "name" => "available", ...
  }
}
```

Example from https://github.com/brightbox/fog/blob/master/lib/fog/aws/models/compute/server.rb#L47

I summoned @solnic on the other issue which was more about dependencies so created this for more of a general discussion.

Basically I'm assuming Virtus can help us if we can work around incompatibilities and/or combine ROM to map between the Hash version of an API response to a model?

I suspect the answer will be "yes, with work" but since I distracted him I thought I'd frame the idea better and see if it's worth me starting a proof of concept.

/cc @elight

Discussion moved from #2
